packages/next/src/server/stream-utils/node-web-streams-helper.test.ts

```diff
@@ -179,7 +180,26 @@ describe('node-web-stream-helpers', () => {
   })
 
   describe('createBufferedTransformStream', () => {})
-  describe('createInsertedHTMLStream', () => {})
+  describe('createInsertedHTMLStream', () => {
+    it('should insert data into the start of the transform', async () => {
+      const encoder = new TextEncoder()
+      function getServerInsertedHTML() {
+        return Promise.resolve('<foo>bar</foo>')
+      }
+      const input = new ReadableStream({
+        start(controller) {
+          controller.enqueue(encoder.encode('<fuzz>buzz</fuzz>'))
```

Comment:
I'd enqueue more than one chunk. what we expect is to reproduce the server inserted html on each chunk. This currently makes sense because we always precede this transform with one that creates a buffered chunk of an entire set of HTML that React is trying to flush. If we got rid of the giant buffer of html then this implementation would no longer serve it's purpose.

This stream helper's implementation is so simple I'm not sure it warrants testing at this level since the test doesn't really encode the semantic. There are more complicated streams helpers where these types of low level tests are necessary to fully exercise all the edge cases being handled within. I think you can land this if you think it's a good idea but I don't think you should feel compelled to